packages/bundlers/default/src/DefaultBundler.js

```diff
@@ -373,6 +437,89 @@ function createIdealGraph(
   let assets = [];
 
   let typeChangeIds = new Set();
+  function makeManualAssetToConfigLookup() {
+    let parentsToConfig = new DefaultMap(() => []);
+
+    for (let c of config.manualSharedBundles) {
+      if (c.parent != null) {
+        parentsToConfig.get(path.join(config.projectRoot, c.parent)).push(c);
+      }
+    }
+    let numParentsToFind = parentsToConfig.size;
+    let configToParentAsset = new Map();
+
+    assetGraph.traverse((node, _, actions) => {
+      if (node.type === 'asset' && parentsToConfig.has(node.value.filePath)) {
+        for (let c of parentsToConfig.get(node.value.filePath)) {
+          configToParentAsset.set(c, node.value);
+        }
+
+        numParentsToFind--;
+
+        if (numParentsToFind === 0) {
+          // If we've found all parents we can stop traversal
+          actions.stop();
+        }
+      }
+    });
+
+    let manualAssetToConfig = new Map();
+    let constantModuleToMSB = new DefaultMap(() => []);
+
+    // Process in reverse order so earlier configs take precedence
+    for (let c of config.manualSharedBundles.reverse()) {
+      invariant(
+        c.parent == null || configToParentAsset.has(c),
+        'Invalid manual shared bundle. Could not find parent asset.',
+      );
+
+      let parentAsset = configToParentAsset.get(c);
+
+      assetGraph.traverse((node, _, actions) => {
+        if (
+          node.type === 'asset' &&
+          (!Array.isArray(c.types) || c.types.includes(node.value.type)) &&
+          isGlobMatch(
```

Comment:
A warning with `isGlobMatch` I found with the lazy selective compilation work - `isGlobMatch` will compile the glob to a regex for every invocation - this can get expensive as well as result in a lot of objects to be GCd.

It might be better to take the approach I did for lazy mode which is that the resolved config is an `Array<RegExp>`, and the config resolver calls `globToRegex` from utils. this code then becomes something like `c.assets.some(r => r.test(node.value.filePath.slice(config.projectRoot.length + 1)))`

Reply:
Yep good point. Will have a look.

Reply:
Done